### PR TITLE
add file extension params with defaults

### PR DESCRIPTION
**Pull request by ChanceNCounter**
defaults include .jsondb, .json and .conf
JsonConfigXDG is the only one that's changed from the previous,
  hardcoded default

### CODE REVIEW

File: json_database/__init__.py [hunk 1]
```diff
@@ -104,10 +104,14 @@ def __exit__(self, _type, value, traceback):
 
 class JsonDatabase(dict):
     """ searchable persistent dict """
-    def __init__(self, name, path=None, disable_lock=False):
+    def __init__(self,
+            name,
+            path=None,
+            disable_lock=False,
+            extension=".json"):
```
Comment (JarbasAl) [inline json_database/__init__.py:111]: i think most people will get tripped up with passing the `.` , maybe the argument should take only the extension itself and the `.` handled internally?

Comment (ChanceNCounter) [reply json_database/__init__.py:111]: good point

File: json_database/__init__.py [hunk 2]
```diff
@@ -278,24 +282,29 @@ def search_by_value(self, key, value, fuzzy=False, thresh=0.7):
...
 
 
 class JsonDatabaseXDG(JsonDatabase):
     """ xdg respectful json database """
 
     def __init__(self, name, xdg_folder=BaseDirectory.xdg_data_home,
-                 disable_lock=False, subfolder="json_database"):
-        path = join(xdg_folder, subfolder, name + ".jsondb")
+                 disable_lock=False, subfolder="json_database",
+                 extension=".jsondb"):
```
Comment (JarbasAl) [inline json_database/__init__.py:300]: remove the `.`

File: json_database/__init__.py [hunk 3]
```diff
@@ -104,10 +104,14 @@ def __exit__(self, _type, value, traceback):
...
-    def __init__(self, name, path=None, disable_lock=False):
+    def __init__(self,
+            name,
+            path=None,
+            disable_lock=False,
+            extension="json"):
         super().__init__()
         self.name = name
-        self.path = path or self.name + ".json"
+        self.path = path or self.name + extension
```
Comment (JarbasAl) [inline json_database/__init__.py:114]: replace with `f"{name}.{extension}"`

File: json_database/__init__.py [hunk 4]
```diff
@@ -278,24 +282,29 @@ def search_by_value(self, key, value, fuzzy=False, thresh=0.7):
...
 class JsonDatabaseXDG(JsonDatabase):
     """ xdg respectful json database """
 
     def __init__(self, name, xdg_folder=BaseDirectory.xdg_data_home,
-                 disable_lock=False, subfolder="json_database"):
-        path = join(xdg_folder, subfolder, name + ".jsondb")
+                 disable_lock=False, subfolder="json_database",
+                 extension=".jsondb"):
+        path = join(xdg_folder, subfolder, f"{name}.{extension}")
         super().__init__(name, path, disable_lock=disable_lock)
```
Comment (JarbasAl) [inline json_database/__init__.py:302]: extension should also be passed to super

File: json_database/__init__.py (hunk 2 above)
Comment (ChanceNCounter) [reply json_database/__init__.py:300]: findfail 🙈

File: json_database/__init__.py (hunk 3 above)
Comment (ChanceNCounter) [reply json_database/__init__.py]: think there's a typo in the fix on this one

Comment (ChanceNCounter) [reply json_database/__init__.py]: yep, macrofail to go with the findfail. pushing fix-fix.

**Pull request merged**